Dev lp resize

alariono opened the pull request:
Hello Ben,
Please take a look on the VD+SFC patch for video analytic sample.

yunbiaolin committed:
add OpenCL driver link

alariono committed:
Build fix: corrected path for MediaSDK includes

alariono committed:
Added option for LP resize after decoder

yunbiaolin commented on video_analytics_example/main.cpp:
```diff
@@ -1259,6 +1266,37 @@ int main(int argc, char *argv[])
+                /* need to correct VPP params: re-size done after decoding
+                 * So, VPP for CSC NV12->RGBP only */
+                VPPParams.vpp.In.CropW = VPPParams.vpp.Out.CropW;
+                VPPParams.vpp.In.CropH = VPPParams.vpp.Out.CropH;
```
it is just a patch to configure the decoding to enable VD+SFC, but I didn't see changes to pass the VPP output surface to decoding? because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.

alariono replied:
Hello Ben,
I have added new version of patch, please look.
>> but I didn't see changes to pass the VPP output surface to decoding?
Ok, done.

>> because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.
Actually, this is not strictly required, because application may use big surface with small cropping for processing. For example, surface resolution is 1920x1088, but cropping area is 352x288 only. (of course memory usage become higher is this case)

alariono commented:
Hello Ben,
I have added new version of patch, please look.
>> but I didn't see changes to pass the VPP output surface to decoding?
Ok, done.

>> because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.
Actually, this is not strictly required, because application may use big surface with small cropping for processing. For example, surface resolution is 1920x1088, but cropping area is 352x288 only. (of course memory usage become higher is this case)

Andrey

From: yunbiaolin [mailto:notifications@github.com]
Sent: Tuesday, September 18, 2018 9:41 AM
To: yunbiaolin/video_analytics_Intel_GPU <video_analytics_Intel_GPU@noreply.github.com>
Cc: Larionov, Andrey <andrey.larionov@intel.com>; Author <author@noreply.github.com>
Subject: Re: [yunbiaolin/video_analytics_Intel_GPU] Dev lp resize (#1)


@yunbiaolin commented on this pull request.

________________________________

In video_analytics_example/main.cpp<https://github.com/yunbiaolin/video_analytics_Intel_GPU/pull/1#discussion_r218313319>:

> +                DecoderPostProcessing.Out.Width = MSDK_ALIGN16(VPPParams.vpp.Out.Width);

+                DecoderPostProcessing.Out.Height = MSDK_ALIGN16(VPPParams.vpp.Out.Height);

+                DecoderPostProcessing.Out.CropX = 0;

+                DecoderPostProcessing.Out.CropY = 0;

+                DecoderPostProcessing.Out.CropW = VPPParams.vpp.Out.CropW;

+                DecoderPostProcessing.Out.CropH = VPPParams.vpp.Out.CropH;

+

+                //DecParams.ExtParam = reinterpret_cast<mfxExtBuffer**>(&DecoderPostProcessing);

+                DecParams.ExtParam[0] = (mfxExtBuffer*)&(DecoderPostProcessing);

+                DecParams.NumExtParam = 1;

+                std::cout << "\t.Decoder's post-processing is used for resizing\n"<< std::endl;

+

+                /* need to correct VPP params: re-size done after decoding

+                 * So, VPP for CSC NV12->RGBP only */

+                VPPParams.vpp.In.CropW = VPPParams.vpp.Out.CropW;

+                VPPParams.vpp.In.CropH = VPPParams.vpp.Out.CropH;

it is just a patch to configure the decoding to enable VD+SFC, but I didn't see changes to pass the VPP output surface to decoding? because it needs to return two kinds of surface, one is decoding output, the other one is resized surface.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/yunbiaolin/video_analytics_Intel_GPU/pull/1#pullrequestreview-156220884>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AcgqsXoOmBySPL0SLURF3IyXfqQnbpAoks5ucJV3gaJpZM4WsEl_>.

--------------------------------------------------------------------
Joint Stock Company Intel A/O
Registered legal address: Krylatsky Hills Business Park,
17 Krylatskaya Str., Bldg 4, Moscow 121614,
Russian Federation

This e-mail and any attachments may contain confidential material for
the sole use of the intended recipient(s). Any review or distribution
by others is strictly prohibited. If you are not the intended
recipient, please contact the sender and delete all copies.

yunbiaolin commented on video_analytics_example/main.cpp:
```diff
@@ -1167,7 +1169,12 @@ int main(int argc, char *argv[])
         std::cout << "\t. Start preparing video parameters for decoding." << std::endl;
+        mfxExtDecVideoProcessing DecoderPostProcessing;
         mfxVideoParam DecParams;
+        bool VD_LP_resize = true;
```
default value should be false, right?

alariono replied:
I have set what VDBOX+SFC enabled by default. My impression was what we need to enable VDBOX+SFC by degault...
But if need i can change behavior to disable VDBOX+SFC mode by default, of course.

alariono committed:
Added VDBOX+SFC support code path; new option "-dec_postproc <>"
Default is auto (-dec_postproc auto). If possible application use VDBOX+SFC path.
Use option "-dec_postproc off" for old behavior (separate decoder + vpp)

Signed-off-by: Andrey Larionov <andrey.larionov@intel.com>

alariono commented:
Hello Ben,
I have added CLI option for VDBOX+SFC case per your request.
